app/controllers/students_controller.rb

```diff
@@ -1,9 +1,58 @@
 class StudentsController < ApplicationController
     before_action :authenticate_user!
     before_action :set_student, only: %i[ show edit update destroy ]
-    # GET /students
+    # GET /student
     def index
-      @students = Student.where(teacher: current_user.email)
+	@students = Student.where(teacher: current_user.email)
+	@tags = Set[]
+	@emails = Set[]
+
+	@courses_taken = Hash[]
+	@semesters_taken = Hash[]
+	for student in @students do
+		@tags.add(student.tags)
+
+		# Figure out each student's course/semester they have taken
+		@courses_taken[student.course_id] = Course.find(student.course_id).course_name
+		@semesters_taken[student.course_id] = Course.find(student.course_id).semester
+
+	end
+	@semesters = Set[]
+	@sections = Set[]
+	@course_names = Set[]
+        @course_ids = Array[]
+		for record in Course.all do
+			@semesters.add(record.semester)
+			@sections.add(record.section)
+			@course_names.add(record.course_name)
+            @course_ids.append(record.id)
+		end
+
+        if params[:selected_course].nil? == false or params[:selected_semester].nil? == false or params[:selected_tag].nil? == false 
+            #dropdown menu selections
+            @selected_tag = params[:selected_tag]
+            @selected_semester = params[:selected_semester]
+            @selected_course = params[:selected_course]
+            #get all course id's for the selected semester + course combo
+            if @selected_semester == '' and @selected_course != ''
+                @target_course_id = Course.where(course_name: @selected_course)
+            elsif @selected_semester != '' and @selected_course == ''
+                @target_course_id = Course.where(semester: @selected_semester)
+            elsif @selected_semester != '' and @selected_course != ''
+                @target_course_id = Course.where(course_name: @selected_course, semester: @selected_semester)
+                #nil check in case the selected semester doesn't have the selected course
+                if @target_course_id.length > 0
```

Comment:
indentation issue

Reply:
the if-statement on line 44 is meant to be inside the elsif on line 41, so this is correct

Reply:
but should check be for all though?

Reply:
target_course_id could be nil in other cases too

Reply:
No since dropdown menu options will only show semesters/courses that are already present in the db. 

Turns out that we don't need this check at all lol. We can pass an active record object directly as a parameter into where(), which means we don't need to manually do the [0].id, so this statement isn't needed. And in the case where target_course_id is an empty array, Student.where() returns an empty array. So no nil check needed.

Reply:
okay cool